R-package/tests/testthat/test_learning_to_rank.R

```diff
@@ -48,14 +49,17 @@ test_that("learning-to-rank with lgb.train() works as expected", {
     }
     expect_identical(sapply(eval_results, function(x) {x$name}), eval_names)
     expect_equal(eval_results[[1L]][["value"]], 0.775)
-    if (!ON_SOLARIS) {
+    if (!(ON_SOLARIS || ON_32_BIT_WINDOWS)) {
```

Comment:
maybe it is better to relax the constraint values? in case there are more tests may fail

Reply:
do you mean change the numeric tolerance?

I think it's better to just completely skip these tests for now to be sure we don't get kicked off of CRAN or waste their time with too many submissions. I don't know the exact amount that the values are different by, and I'm not sure I can figure that out in a way that's guaranteed to be the same as CRAN's environment.

By the next LightGBM release, we can try to completely fix this instead of just skipping the tests.

To be clear, in GitHub Actions here these tests are still running on Linux, Mac, and 64-bit Windows, with multiple different compilers, and for R 3.6 and R 4.0. So this feature is still being thoroughly tested.